### PR TITLE
[AzureMonitorExporter] Add Az.Namespace to Dependency Type.

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
TODO: In a follow up PR cover tests.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/RemoteDependencyData.cs [hunk 1]
```diff
@@ -71,6 +71,8 @@ public RemoteDependencyData(int version, Activity activity, ref ActivityTagsProc
                 Type = "InProc";
             }
 
+            Type = activityTagsProcessor.UnMappedTags.GetAzNameSpace() ?? Type;
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/RemoteDependencyData.cs:74]: could this statement come before line 69 so we're not setting Type twice?

**Comment by TimothyMothra** [comment]
> TODO: In a follow up PR cover tests.

This is a small PR. Why not add tests here?

**Comment by rajkumar-rangaraj** [comment]
> > TODO: In a follow up PR cover tests.
> 
> This is a small PR. Why not add tests here?

PR is small, impact it brings is huge. At this point, we are not sure if this might gets merged or reverted later.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/RemoteDependencyData.cs [hunk 2]
```diff
@@ -66,13 +66,13 @@ public RemoteDependencyData(int version, Activity activity, ref ActivityTagsProc
                     break;
             }
 
-            if (activity.Kind == ActivityKind.Internal && activity.Parent != null)
+            Type = activityTagsProcessor.UnMappedTags.GetAzNameSpace() ?? Type;
```
Comment (TimothyMothra) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/RemoteDependencyData.cs:69]: This change would still allow Type to be set twice.

Stepping back... which should take precidence; AzNameSpace or ActivityKind?

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/RemoteDependencyData.cs:69]: To better explain, might need to give a context here. 

For example, when a storage blob is created using Azure Blobs SDK it creates two spans. One span is marked as kind Internal and other as Client. For the span with kind client we need to set the value of `az.namespace` and for the other span with internal kind we need to value as InProc. This way everything works in AppMap.

```
{"name":"RemoteDependency","time":"2023-04-18T17:36:31.3931090Z","sampleRate":100,"iKey":"00000000-0000-0000-0000-000000000000","tags":{"ai.operation.parentId":"386eaa2913c9302f","ai.operation.id":"88762bac3816b5e9f4c834a7f4eb42b5","ai.user.userAgent":"azsdk-net-Storage.Blobs/12.16.0 (.NET 7.0.5; Microsoft Windows 10.0.22621)","ai.cloud.role":"[my-namespace]/my-service","ai.cloud.roleInstance":"my-instance","ai.internal.sdkVersion":"dotnet7.0.5:otel1.4.0:ext1.0.0-alpha.20230418.1"},"data":{"baseType":"RemoteDependencyData","baseData":{"id":"11dc5d1ae415cf54","name":"PUT /sample-container-test1","resultCode":"201","data":"https://testblob.blob.core.windows.net/sample-container-test1?restype=container","type":"Microsoft.Storage","target":"testblob.blob.core.windows.net","duration":"00:00:01.2209328","success":true,"properties":{"az.client_request_id":"a8aad3b8-ea79-405f-8096-a7301f50dfab","az.namespace":"Microsoft.Storage","az.schema_url":"https://opentelemetry.io/schemas/1.17.0","az.service_request_id":"815fdd0d-801e-004a-041c-7223a9000000","CustomDimension1":"Value1","CustomDimension2":"Value2","ActivityKind":"Client"},"ver":2}}}
{"name":"RemoteDependency","time":"2023-04-18T17:36:31.3324025Z","sampleRate":100,"iKey":"00000000-0000-0000-0000-000000000000","tags":{"ai.operation.id":"88762bac3816b5e9f4c834a7f4eb42b5","ai.cloud.role":"[my-namespace]/my-service","ai.cloud.roleInstance":"my-instance","ai.internal.sdkVersion":"dotnet7.0.5:otel1.4.0:ext1.0.0-alpha.20230418.1"},"data":{"baseType":"RemoteDependencyData","baseData":{"id":"386eaa2913c9302f","name":"Enriched-BlobContainerClient.Create","type":"InProc","duration":"00:00:01.2950693","success":true,"properties":{"az.namespace":"Microsoft.Storage","az.schema_url":"https://opentelemetry.io/schemas/1.17.0","CustomDimension1":"Value1","CustomDimension2":"Value2","ActivityKind":"Internal"},"ver":2}}}
```

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/RemoteDependencyData.cs:69]: Could you please add a comment explaining this? that way anyone can understand why we are doing this.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/RemoteDependencyDataTests.cs [hunk 3]
```diff
@@ -57,25 +57,32 @@ public void ValidateDBDependencyType(string dbSystem)
...
+            activity?.AddTag("az.namespace", "DemoAzureResource");
 
-            var remoteDependencyDataTypeForChild = new RemoteDependencyData(2, childActivity, ref activityTagsProcessorChild).Type;
+            Assert.NotNull(activity);
+            var activityTagsProcessor = TraceHelper.EnumerateActivityTags(activity);
+
+            var remoteDependencyDataType = new RemoteDependencyData(2, activity, ref activityTagsProcessor).Type;
 
-            Assert.Equal("InProc", remoteDependencyDataTypeForChild);
+            Assert.Equal("DemoAzureResource", remoteDependencyDataType);
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/RemoteDependencyDataTests.cs:85]: Add a check for custom property too (az.namespace will be available on it)

**Review by vishweshbankwar — APPROVED** [review]
LGTM with minor suggestions.

**Pull request merged**